rider-core/src/main/java/com/github/database/rider/core/dataset/DataSetExecutorImpl.java

```diff
@@ -861,7 +883,90 @@ public void compareCurrentDataSetWith(DataSetConfig expectedDataSetConfig, Strin
+                sbFacts.append(
+                        Arrays.stream(actualTable.getTableMetaData().getColumns())
+                                .map(column -> {
+                                    try {
+                                        return finalActualTable.getValue(finalI, column.getColumnName());
+                                    } catch (DataSetException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                })
+                                .map(o -> "'" + o + "'")
+                                .collect(Collectors.joining(","))
+                );
+                sbFacts.append(").\n");
+            }
+
+            for (int i = 0; i < expectedTable.getRowCount(); i++) {
+                StringBuilder sbQuery = new StringBuilder();
+                sbQuery.append(expectedTable.getTableMetaData().getTableName().toLowerCase());
+                sbQuery.append("(");
+                int finalI = i;
+                ITable finalExpectedTable = expectedTable;
+                sbQuery.append(
+                        Arrays.stream(actualTable.getTableMetaData().getColumns())
+                                .map(column -> {
+                                    try {
+                                        return finalExpectedTable.getValue(finalI, column.getColumnName());
+                                    } catch (DataSetException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                })
+                                // FIXME: simple #toString() may not work well with all datatypes
+                                .map(o -> o == null ? "_" : (o.toString().startsWith("$$") && o.toString().endsWith("$$")) ? o.toString().replaceAll("\\$\\$", "").toUpperCase() : "'" + o + "'")
+                                .collect(Collectors.joining(","))
+                );
+                sbQuery.append(")");
+                queryTerms.add(sbQuery.toString());
+            }
+        }
+
+        log.info("Facts = " + sbFacts);
+
+        log.info("Query = " + String.join(",", queryTerms));
+
+        ClausesReader theoryReader = ClausesReader.getWithDefaultOperators();
+        SolverFactory solverFactory = ClassicSolverFactory.INSTANCE; // or Solver.getClassic()
+
+        ByteArrayInputStream bais = new ByteArrayInputStream(
+                sbFacts.toString().getBytes(StandardCharsets.UTF_8)
+        );
+
+        Theory theory = theoryReader.readTheory(bais);
+        Solver solver = solverFactory.solverWithDefaultBuiltins(theory);
+
+        TermParser termParser = TermParser.withOperators(solver.getOperators());
+
+        it.unibo.tuprolog.core.Struct query = termParser.parseStruct(String.join(",", queryTerms));
+
+        Iterator<Solution> si = solver.solve(query, SolveOptions.allLazilyWithTimeout(1000)).iterator();
```

Comment:
Is this timeout enough for large datasets?  shall we move this to [dbunit configuration](https://github.com/database-rider/database-rider#332-dbunit-configuration)?

Reply:
Sure, makes sense